pkg/services/github.go

```diff
@@ -31,49 +31,51 @@ type GitHubOptions struct {
 }
 
 type GitHubNotification struct {
-	repoURL  string
-	revision string
-	Status   *GitHubStatus `json:"status,omitempty"`
-}
-
-type GitHubStatus struct {
 	State     string `json:"state,omitempty"`
 	Label     string `json:"label,omitempty"`
 	TargetURL string `json:"targetURL,omitempty"`
+	RepoURL   string `json:"repoURL,omitempty"`
+	Revision  string `json:"revision,omitempty"`
 }
 
 const (
-	repoURLtemplate  = "{{.app.spec.source.repoURL}}"
-	revisionTemplate = "{{.app.status.operationState.syncResult.revision}}"
+	defaultRepoURLtemplate  = "{{.app.spec.source.repoURL}}"
+	defaultRevisionTemplate = "{{.app.status.operationState.syncResult.revision}}"
 )
 
 func (g *GitHubNotification) GetTemplater(name string, f texttemplate.FuncMap) (Templater, error) {
+
+	repoURLtemplate := defaultRepoURLtemplate
+	if g.RepoURL != "" {
+		repoURLtemplate = g.RepoURL
```

Comment:
I think `g.RepoURL` refers to the actual git repo URL (e.g. `https://github.com/argoproj-labs/argocd-notifications.git`), while `repoURLtemplate` refers to the API field path (e.g. `{{.app.spec.source.repoURL}}`).
I would suggest using two fields to separate them as #115.